File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program66.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept N numbers from user and return difference between summation of even elements and summation of odd elements.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Solution
{
	public int iSize,arr[],iCnt,iDiff,iEven,iOdd;

	public Solution(int iNo)
	{
		this.iDiff = 0;
		this.iEven = 0;
		this.iOdd = 0;
		this.iCnt = 0;
		this.iSize = iNo;
		arr = new int[iSize];
	}

	public void Accept()
	{
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter "+arr.length+" elements: ");
		for(iCnt=0;iCnt<arr.length;iCnt++)
		{
			System.out.print("Enter "+ (iCnt+1) + "element: ");
			arr[iCnt] = sObj.nextInt();
		}
	}	

	public int Display()
	{
		for(iCnt=0;iCnt<arr.length;iCnt++)
		{
			if(arr[iCnt]%2==0)
			{
				iEven = iEven + arr[iCnt];
			}
			else
			{
				iOdd = iOdd + arr[iCnt];
			}
		}

		iDiff = iEven-iOdd;
		return iDiff;
	}

}

class Program
{
	public static void main(String[] args)
	{
		int iRet = 0;
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter number of elements:");
		int iLength = sObj.nextInt();

		Solution obj = new Solution(iLength);
		obj.Accept();
		iRet = obj.Display();
	
		System.out.println("Result is: "+iRet);
	}
}

/*
OUTPUT:
Enter number of elements:
6
Enter 6 elements: 
Enter 1element: 85
Enter 2element: 66
Enter 3element: 3
Enter 4element: 80
Enter 5element: 93
Enter 6element: 88
Result is: 53

*/

File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program66.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept N numbers from user and return difference between summation of even elements and summation of odd elements.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class DiffEvenOdd
{
	private:
		int *iArr;
		int iSize,iCnt=0,iEven=0,iOdd=0,iDiff=0;

	public:
		DiffEvenOdd(int iNo=0,int iCnt=0,int iEven=0,int iOdd=0,int iDiff=0)
		{
			this->iSize=iNo;
			iArr = new int[iSize];
		}

		void Accept()
		{
			cout<<"Enter "<<iSize<<" elements:\n";
			for(iCnt=0;iCnt<iSize;iCnt++)
			{
				cout<<"Enter "<<iCnt+1<<" element: ";
				cin>>iArr[iCnt];
			}
		}
		
		int Difference()
		{
			for(iCnt=0;iCnt<iSize;iCnt++)
			{
				if(*(iArr+iCnt)%2==0)
				{
					iEven = iEven+ *(iArr+iCnt);
				}
				else
				{
					iOdd = iOdd+ *(iArr+iCnt);
				}	
			}

			iDiff = iEven-iOdd;
			return iDiff;
		}

		~DiffEvenOdd()
		{
			delete []iArr;
		}

};

int main()
{
	int iLength=0,iRet=0;

	cout<<"Enter number of elements: ";
	cin>>iLength;

	DiffEvenOdd *dObj = new DiffEvenOdd(iLength);

	dObj->Accept();
	iRet = dObj->Difference();

	cout<<"Result is: "<<iRet<<"\n";

	delete dObj;
	return 0;
}

/*
OUTPUT:

Enter number of elements: 6
Enter 6 elements:
Enter 1 element: 85
Enter 2 element: 66
Enter 3 element: 3
Enter 4 element: 80
Enter 5 element: 93
Enter 6 element: 88
Result is: 53

*/
